runtime/lua/vim/lsp/util.lua

```diff
@@ -962,7 +968,10 @@ function M.preview_location(location)
   local range = location.targetRange or location.range
   local contents = api.nvim_buf_get_lines(bufnr, range.start.line, range["end"].line+1, false)
   local syntax = api.nvim_buf_get_option(bufnr, 'syntax')
-  return M.open_floating_preview(contents, syntax)
+  if syntax == "" then
+    syntax = api.nvim_buf_get_option(bufnr, 'filetype')
```

Comment:
I think there are cases filetype != syntax, does this work with typescriptreact, ocaml, etc?

Reply:
yep, there was a todo about this already in the fancy_floating_window code. But in this case it only used ft if there's no syntax set.

So worst case it won't use any syntax

Reply:
I'd put a comment here at least :)

Reply:
ok! :)